UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/HealthCheck/AzureBlobStorageHealthCheck.cs

```diff
@@ -0,0 +1,81 @@
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Logging;
+using UKHO.ExchangeSetService.Common.Storage;
+
+namespace UKHO.ExchangeSetService.Common.HealthCheck
+{
+    public class AzureBlobStorageHealthCheck : IHealthCheck
+    {
+        private readonly IAzureBlobStorageClient azureBlobStorageClient;
+        private readonly ISalesCatalogueStorageService scsStorageService;
+        private readonly IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration;
+        private readonly ILogger<AzureBlobStorageService> logger;
+
+        public AzureBlobStorageHealthCheck(IAzureBlobStorageClient azureBlobStorageClient,
+                                           ISalesCatalogueStorageService scsStorageService,
+                                           IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration,
+                                           ILogger<AzureBlobStorageService> logger)
+        {
+            this.azureBlobStorageClient = azureBlobStorageClient;
+            this.scsStorageService = scsStorageService;
+            this.essFulfilmentStorageConfiguration = essFulfilmentStorageConfiguration;
+            this.logger = logger;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                string[] exchangeSetTypes = essFulfilmentStorageConfiguration.Value.ExchangeSetTypes.Split(",");
+                string storageAccountConnectionString = string.Empty;
+                HealthCheckResult azureBlobStorageHealthStatus = new HealthCheckResult(HealthStatus.Healthy);
+                foreach (string exchangeSetType in exchangeSetTypes)
+                {
+                    var storageAccountWithKey = GetStorageAccountNameAndKey(exchangeSetType);
+                    storageAccountConnectionString = scsStorageService.GetStorageAccountConnectionString(storageAccountWithKey.Item1, storageAccountWithKey.Item2);
+                    azureBlobStorageHealthStatus = await azureBlobStorageClient.CheckBlobContainerHealth(storageAccountConnectionString, essFulfilmentStorageConfiguration.Value.StorageContainerName);
+                    if (azureBlobStorageHealthStatus.Status == HealthStatus.Unhealthy)
+                        break;
+                }
+                if (azureBlobStorageHealthStatus.Status == HealthStatus.Healthy)
+                {
+                    logger.LogDebug(EventIds.AzureBlobStorageIsHealthy.ToEventId(), $"Azure blob storage is healthy");
+                    return HealthCheckResult.Healthy("Azure blob storage is healthy");
+                }
+                else
+                {
+                    logger.LogError(EventIds.AzureBlobStorageIsUnhealthy.ToEventId(), $"Azure blob storage is unhealthy with error {azureBlobStorageHealthStatus.Exception.Message}");
```

Comment:
Kindly use structured logging rather than interpolated string on all log messages.
Also log the exception using the LogError overload that takes the exception as an explicit parameter for all logerror.

Also EventId is missing at line 60

Reply:
Fixed